docs/linux/setup_linux-host_qemu-vm_riscv64-kernel.md

```diff
@@ -0,0 +1,149 @@
+["riscv: Allow building with kcov coverage"](https://lore.kernel.org/linux-riscv/20200626124056.29708-1-tklauser@distanz.ch/)
+patch. Create a kernel config with:
+
+```shell
+make ARCH=riscv CROSS_COMPILE=riscv64-linux-gnu- defconfig
+```
+
+Also enable the [recommended Kconfig options for syzkaller](/docs/linux/kernel_configs.md).
+
+Then build kernel with:
+
+```
+make ARCH=riscv CROSS_COMPILE=riscv64-linux-gnu- -j $(nproc) vmlinux
+```
+
+# BBL (Berkeley Boot Loader)
+
+Clone the BBL repository and build the bootable image containg bootloader and kernel:
+
+```shell
+git clone https://github.com/riscv/riscv-pk
+cd riscv-pk
+mkdir build && cd build
+../configure --enable-logo --host=riscv64-linux-gnu --with-payload=/path/to/linux/vmlinux
+make
+```
+
+# Image
+
+We will use buildroot to create the disk image. You can obtain buildroot
+[here](https://buildroot.uclibc.org/download.html). The following instructions
+were tested with buildroot version 2020.05. First run:
+
+```shell
+make qemu_riscv64_virt_defconfig
+make menuconfig
+```
+
+Choose the following options:
+
+```
+    Target packages
+	    Networking applications
+	        [*] dhcpcd
+	        [*] iproute2
+	        [*] openssh
+    Filesystem images
+                ext2/3/4 variant - ext4
+	        exact size - 1g
+```
+
+Unselect:
+
+```
+    Kernel
+	    Linux Kernel
+```
+
+Run `make`.
+
+Then add the following line to `output/target/etc/fstab`:
+
+```
+debugfs	/sys/kernel/debug	debugfs	defaults	0	0
+```
+
+Then replace `output/target/etc/ssh/sshd_config` with the following contents:
+
+```
+PermitRootLogin yes
+PasswordAuthentication yes
+PermitEmptyPasswords yes
+```
+
+Run `make` again.
+
+# Test kernel and image
+
+Run:
+
+```shell
+qemu-system-riscv64 \
+	-machine virt \
+	-nographic \
+	-kernel /riscv-pk/build/bbl \
+	-append "root=/dev/vda ro console=ttyS0" \
+	-object rng-random,filename=/dev/urandom,id=rng0 \
+	-device virtio-rng-device,rng=rng0 \
+	-drive file=/buildroot/output/images/rootfs.ext2,if=none,format=raw,id=hd0 \
+	-device virtio-blk-device,drive=hd0 \
+	-netdev user,id=net0,host=10.0.2.10,hostfwd=tcp::10022-:22 \
+	-device virtio-net-device,netdev=net0
+```
+
+This should boot the kernel. Wait for login prompt, then in another console run:
+
+```
+ssh -p 10022 root@localhost
+```
+
+ssh should succeed.
+
+# syzkaller
+
+Build syzkaller as described [here](/docs/linux/setup.md#go-and-syzkaller), with `riscv64` target:
+
+```
+make TARGETOS=linux TARGETARCH=riscv64
+```
+
+Create the manager config `riscv64.cfg` similar to the following one (adjusting paths as necessary):
+
+```
+{
+	"name": "riscv64",
+	"target": "linux/riscv64",
+	"http": ":56700",
+	"workdir": "/workdir",
+	"kernel_obj": "/linux",
+	"syzkaller": "/gopath/src/github.com/google/syzkaller",
+	"image": "/buildroot/output/images/rootfs.ext2",
+	"procs": 8,
+	"type": "qemu",
+	"vm": {
+		"count": 1,
+		"kernel": "/riscv-pk/build/bbl",
+		"cpu": 2,
```

Comment:
I am getting:
```
qemu-system-riscv64: Invalid SMP CPUs 4. The max CPUs supported by machine 'spike' is 1
2020/06/27 13:56:22 failed to create instance: failed to read from qemu: EOF
```

Reply:
Looks like you need `-machine virt` in `qemu_args`, but this is already the default in `archConfig` now :thinking: Did you override `qemu_args` in `riscv64.cfg`?